src/OpenIddict.EF/OpenIddictContext.cs

```diff
@@ -10,29 +10,120 @@
 using OpenIddict.Models;
 
 namespace OpenIddict {
-    public class OpenIddictContext<TUser, TApplication, TRole, TKey> : IdentityDbContext<TUser, TRole, TKey>
+    /// <summary>
+    /// Base class for the Entity Framework database context used for OpenIddict.
+    /// </summary>
+    /// <typeparam name="TApplication">The type of application objects.</typeparam>
+    /// <typeparam name="TApplicationKey">The type used for the primary key for the application.</typeparam>
+    /// <typeparam name="TUser">The type of user objects.</typeparam>
+    /// <typeparam name="TRole">The type of role objects.</typeparam>
+    /// <typeparam name="TKey">The type of the primary key for users and roles.</typeparam>
+    /// <seealso cref="Microsoft.AspNetCore.Identity.EntityFrameworkCore.IdentityDbContext{TUser, TRole, TKey}" />
+    public class OpenIddictContext<TApplication, TApplicationKey, TUser, TRole, TKey> : IdentityDbContext<TUser, TRole, TKey>
```

Comment:
> What if I want to use a Guid as a key for Identity and string as an ApplicationId and key at the same time?

You won't be able to do that, just like Identity doesn't allow you to use different key types for user/role. With the same logic, we'd have to add a `TAuthorizationKey`, a `TScopeKey`, a `TTokenKey` and honestly, it would be a real nightmare.

Don't forget that Identity added new generic arguments recently: https://github.com/aspnet/Identity/pull/798/files#diff-c72544a50eacab8ada8ca53f53daee15R84.

That would be great if we could avoid having 20 or 30 generic arguments :fearful: